### PR TITLE
Fix various problems indicated by coverity

### PR DESCRIPTION
**Pull request by MarkusFrankATcernch**
BEGINRELEASENOTES
- Fix some coverity problems

ENDRELEASENOTES

### CODE REVIEW

File: DDG4/src/Geant4Kernel.cpp [hunk 1]
```diff
@@ -94,7 +94,7 @@ Geant4Kernel::Geant4Kernel(Detector& description_ref)
   declareProperty("OutputLevel",          m_outputLevel = DEBUG);
   declareProperty("NumEvents",            m_numEvent = 10);
   declareProperty("OutputLevels",         m_clientLevels);
-  declareProperty("NumberOfThreads",      m_numThreads);
+  declareProperty("NumberOfThreads",      m_numThreads = 0);
```
Comment (andresailer) [inline DDG4/src/Geant4Kernel.cpp:97]: Should this be 1, or does it not matter?

Comment (MarkusFrankATcernch) [reply DDG4/src/Geant4Kernel.cpp:97]: I think for single threading it does not matter. For multi threading it should be set properly.

Comment (MarkusFrankATcernch) [reply DDG4/src/Geant4Kernel.cpp:97]: It is set in the constructor explicitly to "0", so in principle the re-assignment in `declareProperty(...)` is redundant.

Comment (andresailer) [reply DDG4/src/Geant4Kernel.cpp:96]: Ok, thanks!

**Pull request merged**